src/Core/src/Platform/iOS/MauiView.cs

```diff
@@ -90,12 +148,34 @@ public ICrossPlatformLayout? CrossPlatformLayout
 
 		Size CrossPlatformMeasure(double widthConstraint, double heightConstraint)
 		{
-			return CrossPlatformLayout?.CrossPlatformMeasure(widthConstraint, heightConstraint) ?? Size.Zero;
+			if (_appliesSafeAreaAdjustments)
+			{
+				// When responding to safe area, we need to adjust the constraints to account for the selective safe area.
+				var selectiveSafeArea = GetSelectiveSafeArea();
+				widthConstraint -= selectiveSafeArea.HorizontalThickness;
+				heightConstraint -= selectiveSafeArea.VerticalThickness;
+			}
+
+			var crossPlatformSize = CrossPlatformLayout?.CrossPlatformMeasure(widthConstraint, heightConstraint) ?? Size.Zero;
+
+			if (_appliesSafeAreaAdjustments)
+			{
+				// If we're responding to the safe area, we need to add the safe area back to the size so the container can allocate the correct space
+				var selectiveSafeArea = GetSelectiveSafeArea();
+				crossPlatformSize = new Size(crossPlatformSize.Width + selectiveSafeArea.HorizontalThickness, crossPlatformSize.Height + selectiveSafeArea.VerticalThickness);
+			}
+
+			return crossPlatformSize;
```

Comment:
Can you analyze these changes and indicate if you agree with what this is doing and the purpose for it? 

Highlight any issues or misunderstandings of how this might work with the iOS layout system